helpers.R

```diff
@@ -145,29 +150,37 @@ where_tissue_pattern_query <- function(pattern) {
 
 get_results_data_from_db <- function(input) {
     # Construct the fetching query
-    #where <- " WHERE m.pval IS NOT null and p.hidden != 1" #make it friendlier to sqlite
+    #where <- " WHERE m.pval IS NOT null and p.hidden != 1" # make it friendlier to sqlite
     where <- " WHERE m.pval IS NOT null and p.hidden != true"
-
-    if (nchar(input$gene_name) > 0){
-      s <- naive_string_sanitation(input$gene_name)
-      s <- get_gene_name_field_conversion(s)
-      where <- paste0(where, " AND g.gene_name LIKE '", s, "%'")
-    }
-
-    if (nchar(input$pheno_pattern) >0) {
-      where <- paste0(where, where_pheno_pattern_query(input$pheno_pattern))
-    }
-
-    if (nchar(input$tissue_pattern) >0) {
-      where <- paste0(where, where_tissue_pattern_query(input$tissue_pattern))
+    
+    if (length(input$gene_name) != 0) {
+      where <- paste0(where, " AND ")
+      where <- paste0(where, "(g.gene_name = '", input$gene_name[1], "'")
+      if(length(input$gene_name) >= 2){
```

Comment:
I would use a for loop for all cases (`i in 1:length()`) and have an `if (i>1)` to add the ` OR ` clause